FtpProvider/FtpProvider/FtpProvider.fs

```diff
@@ -1,102 +1,161 @@
+                        addTypes(site + dir + "/", nestedType)
+                        yield nestedType :> MemberInfo 
+
+                    for file in files do 
+
+                        let nestedType = ProvidedTypeDefinition(file, Some typeof<obj>)
+
+                        let contentsProperty =                                   
+                                ProvidedProperty("Contents", typeof<obj>,
+                                                   IsStatic=true,
+                                                   GetterCode = 
+                                                        (fun args -> 
+                                                            <@@ 
+                                                                let request = WebRequest.Create(site + file) :?> FtpWebRequest
+                                                    
+                                                                request.Method <- WebRequestMethods.Ftp.DownloadFile
+                                                                request.UseBinary <- useBinary
+                                                                request.Credentials <- new NetworkCredential(user, pwd) 
+                                                                let response = request.GetResponse() :?> FtpWebResponse   // TODO.2 
+                                                    
+                                                                use responseStream = response.GetResponseStream()
+                                                                use reader = new StreamReader(responseStream)
+                                                    
+                                                                let r = reader.ReadToEnd() :> obj
+                                                                r
+                                                                // TODO.
+                                                                // TODO.3
+
+                                                              @@>))
+                        nestedType.AddMember contentsProperty
+
+                        yield nestedType :> MemberInfo 
+                   ]
+                   )
+        let actualType = ProvidedTypeDefinition(asm, nameSpace, typeName, Some typeof<obj>)
+        addTypes(site, actualType)
+        actualType
+
+    let addProvidedStaticParameter nme typ xmldoc = 
+      let p = ProvidedStaticParameter(nme,typ) 
+      p.AddXmlDoc(sprintf xmldoc)
+      p
+
+    let _ = 
+        let a = ProvidedTypeDefinition(asm, nameSpace, "FtpProvider", Some typeof<obj>)
+        a.AddXmlDoc("An FTP Type Provider which lets you 'dot' into directory structures, and then retrieve a file by 'dotting' into the '.Contents' tag.  note: there are no progress updates, so if it's a large file over a slow connection, the only solution is to wait.  Perhaps try a smaller file first to verify.")  // BUG.2 

```

Comment:
It looks like type ``a`` is never used.  it should be deleted and the XmlDoc attached to ``topType``?